cors/handler.go

```diff
@@ -0,0 +1,94 @@
+package cors
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+type preflightInfo struct {
+	allowedMethods []string
+	allowedHeaders []string
+}
+
+type Middleware struct {
+	paths map[string]preflightInfo
+}
+
+func New() *Middleware {
+	return &Middleware{
+		paths: map[string]preflightInfo{},
+	}
+}
+
+func (m *Middleware) Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Access-Control-Allow-Origin", r.Header.Get("Origin"))
```

Comment:
If `origin` is missing it should be `*`? or not present?

Reply:
Since we are not relying on this for any security I guess we could just as well set it to `*`

Reply:
That's why maybe we should consider a blanket implementation instead? We verify the allowed actions on the actual call and if the user sends in more headers than we need I don't see a problem with that?

In that case, maybe we should just don't have a middleware and just have a common handler that is set up by the Endpoint struct in the server?